### PR TITLE
Add examples in doc

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -413,26 +472,81 @@ impl UnixStream {
...
 
     /// Returns the read timeout of this socket.
+    ///
+    /// # Examples
+    ///
+    /// ```rust,no_run
+    /// use unix_socket::UnixStream;
+    ///
+    /// let stream = UnixStream::connect("/path/to/my/socket").unwrap();
+    /// println!("{}", match stream.read_timeout() {
```
Comment (sfackler) [inline src/lib.rs:497]: The examples seem to be inconsistent as to whether they panic or match on the result.

File: src/lib.rs [hunk 2]
```diff
@@ -413,26 +472,81 @@ impl UnixStream {
...
 
     /// Moves the socket into or out of nonblocking mode.
+    ///
+    /// # Examples
+    ///
+    /// ```rust,no_run
+    /// use unix_socket::UnixStream;
+    ///
+    /// let stream = UnixStream::connect("/path/to/my/socket").unwrap();
+    /// stream.set_nonblocking(false).unwrap();
```
Comment (sfackler) [inline src/lib.rs]: I'd change `false` to `true` here since sockets are blocking by default.

Comment (GuillaumeGomez) [inline src/lib.rs]: Ok!

File: src/lib.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/lib.rs:497]: I match when I want to print.

**Comment by sfackler** [comment]
Thanks!

**Pull request merged**